scopes/envs/envs/environment.ts

```diff
@@ -18,4 +32,96 @@ export interface Environment {
   icon?: string;
 
   [key: string]: any; // :TODO need to define an abstract type for service handlers (now using any)
+
+  /**
+   * Returns the Environment descriptor
+   * Required for any task
+   */
+  __getDescriptor?: () => Promise<EnvDescriptor>;
+
+  /**
+   * Returns a schema generator instance
+   */
+  getSchemaExtractor?: (config?: any) => SchemaExtractor;
+}
+
+export interface DependenciesEnv extends Environment {
+  /**
+   * Returns the list of dependencies
+   * Required for any task
+   */
+  getDependencies?: () => VariantPolicyConfigObject | Promise<VariantPolicyConfigObject>;
+}
+
+export interface PackageEnv extends Environment {
+  /**
+   * define the package json properties to add to each component.
+   * Used by `bit link` to augment package.json with new properties
+   */
+  getPackageJsonProps?: () => PackageJsonProps;
+}
+
+export interface LinterEnv extends Environment {
+  /**
+   * Returns & configures the linter to use (ESLint, ...)
+   * Required for `bit lint`
+   */
+  getLinter?: () => Linter;
+}
+
+export interface PreviewEnv extends Environment {
+  /**
+   * Returns a paths to a function which mounts a given component to DOM
+   * Required for `bit start` & `bit build`
+   */
+  getMounter?: () => string;
+
+  /**
+   * Returns a path to a docs template.
+   * Required for `bit start` & `bit build`
+   */
+  getDocsTemplate?: () => string;
+
+  /**
+   * Returns a bundler for the preview.
+   * Required for `bit build` & `bit start`
+   */
+  getBundler?: (context: BundlerContext, transformers: any[]) => Promise<Bundler>;
+}
+
+export interface BuilderEnv extends PreviewEnv {
+  /**
+   * @deprecated Fatal: a breaking API was introduced. Use getBuildPipe() instead.
+   */
+  getPipe?: () => BuildTask[];
+
+  /**
+   * Returns the component build pipeline
+   * Required for `bit build`
+   */
+  getBuildPipe?: (tsconfig?: TsConfigSourceFile) => BuildTask[];
```

Comment:
we now have also `getTagPipe` and `getSnapPipe`, but I can take care of them myself once merged.